src/test/java/com/atlassian/bitbucket/jenkins/internal/scm/extensions/BitbucketPostBuildStatusTest.java

```diff
@@ -0,0 +1,115 @@
+package com.atlassian.bitbucket.jenkins.internal.scm.extensions;
+
+import com.atlassian.bitbucket.jenkins.internal.provider.JenkinsProvider;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import com.atlassian.bitbucket.jenkins.internal.status.BuildStatusPoster;
+import com.google.inject.Injector;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.Revision;
+import hudson.plugins.git.util.Build;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.util.BuildData;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BitbucketPostBuildStatusTest {
+
+    private static final String SERVER_ID = "TestServerID";
+    private static final String SHA1 = "67d71c2133aab0e070fb8100e3e71220332c5af1";
+    @Mock
+    private AbstractBuild build;
+    @Mock
+    private BuildData buildData;
+    @Mock
+    private BuildStatusPoster buildStatusPoster;
+    @Captor
+    private ArgumentCaptor<BitbucketRevisionAction> captor;
+    private BitbucketPostBuildStatus extension;
+    @Mock
+    private Injector injector;
+    @Mock
+    private Jenkins jenkins;
+    @Mock
+    private JenkinsProvider jenkinsProvider;
+    @Mock
+    private Build lastBuild;
+    @Mock
+    private TaskListener listener;
+    @Mock
+    private Run notABuild;
+    @Mock
+    private Revision revision;
+    @Mock
+    private GitSCM scm;
+
+    @Before
+    public void setup() {
+        extension = new BitbucketPostBuildStatus(SERVER_ID, jenkinsProvider);
+        buildData.lastBuild = lastBuild;
```

Comment:
Alarmingly, this appears to be the correct solution to this problem (all other build getter methods require the commit hash, which we are trying to get). I can't conceive of a reason this would be done by a plugin, but the possibility remains so our solution may not be robust enough.